packages/esm-outpatient-app/src/queue-patient-linelists/queue-linelist.resource.tsx

```diff
@@ -43,3 +44,38 @@ export function useProviders() {
     isValidating,
   };
 }
+
+export function usePatientAppointments(patientUuid: string, startDate, abortController: AbortController) {
+  const appointmentsSearchUrl = `/ws/rest/v1/appointments/search`;
+  const fetcher = () =>
+    openmrsFetch(appointmentsSearchUrl, {
+      method: 'POST',
+      signal: abortController.signal,
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: {
+        patientUuid: patientUuid,
+        startDate: startDate,
+      },
+    });
+
+  const { data, error, isLoading, isValidating } = useSWR<AppointmentsFetchResponse, Error>(
+    appointmentsSearchUrl,
+    fetcher,
+  );
+
+  const appointments = data?.data?.length ? data.data : null;
+
+  const upcomingAppointments = appointments
```

Comment:
In the event appointments is null. This will through a null exception we can default `appointments` to empty array if no data is found on line 68